x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/capacity/memory/AutoscalingMemoryAndProcessorInfoService.java

```diff
@@ -22,42 +22,47 @@
 import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.inject.Inject;
 import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.util.Maps;
 import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.monitor.os.OsInfo;
 import org.elasticsearch.xpack.autoscaling.AutoscalingMetadata;
 import org.elasticsearch.xpack.autoscaling.policy.AutoscalingPolicy;
 import org.elasticsearch.xpack.autoscaling.policy.AutoscalingPolicyMetadata;
 
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.Locale;
 import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
 import java.util.SortedSet;
 import java.util.function.Function;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
 import static java.util.stream.Collectors.toUnmodifiableMap;
 
-public class AutoscalingMemoryInfoService {
+public class AutoscalingMemoryAndProcessorInfoService {
     public static final Setting<TimeValue> FETCH_TIMEOUT = Setting.timeSetting(
         "xpack.autoscaling.memory.monitor.timeout",
         TimeValue.timeValueSeconds(15),
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
-    private static final Long FETCHING_SENTINEL = Long.MIN_VALUE;
+    private static final MemoryAndProcessors FETCHING_SENTINEL = new MemoryAndProcessors(Long.MIN_VALUE, Integer.MIN_VALUE);
 
-    private static final Logger logger = LogManager.getLogger(AutoscalingMemoryInfoService.class);
+    private static final Logger logger = LogManager.getLogger(AutoscalingMemoryAndProcessorInfoService.class);
 
-    private volatile Map<String, Long> nodeToMemory = Map.of();
+    private volatile Map<String, MemoryAndProcessors> nodeToMemory = Map.of();
     private volatile TimeValue fetchTimeout;
 
     private final Client client;
     private final Object mutex = new Object();
 
     @Inject
-    public AutoscalingMemoryInfoService(ClusterService clusterService, Client client) {
+    public AutoscalingMemoryAndProcessorInfoService(ClusterService clusterService, Client client) {
```

Comment:
Not something for this PR, but this class is doing the same thing as https://github.com/elastic/elasticsearch/pull/87366/files#diff-d71d08cc6d198f07a28ee2e1179053197941de85851c6797f30a516af523d158R725-R726 in a different way.  So we have two different plugins that both want to know how much memory and how many processors other nodes have and are getting that information in two completely different ways.  It might be nice to consolidate this into core server at some point in the future.  If each node provided its memory and processors when joining the cluster then it could be available in cluster state - effectively like what ML is doing with node attributes but integrated as a first-class feature.